app/src/main/java/rick_and_morty/ui/characters/CharactersViewModel.kt

```diff
@@ -59,6 +59,33 @@ class CharactersViewModel @Inject constructor(
     fun onCharacterSelected(characterID: Int) {
         eventBus.postEvent(NavigateToCharacterDetailsEvent(characterID))
     }
+
+    fun refreshCharacters() {
+        viewModelScope.launch {
+            try {
+                _characters.update { it.copy(isLoading = true) }
+
+                characterRepository.clearCharactersDatabase()
+                page = 1
+
+                val dbCharacters = characterRepository.getCharacters(page)
+                characterRepository.saveCharactersToDatabase(dbCharacters)
```

Comment:
Please do not clear any data before you got a more up-to-date version of the data.
You might end up showing nothing.

This principle should also be applied to the `EpisodesViewModel` and the deletion of the database at app start.